File: backend/src/main/java/com/seoulawake/backend/module/cafe/domain/Address.java
package com.seoulawake.backend.module.cafe.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Address {
	@Column
	private String normalAddress;
	private String roadAddress;

	public Address(String normalAddress, String roadAddress) {
		this.normalAddress = normalAddress;
		this.roadAddress = roadAddress;
	}
}

File: backend/src/main/kotlin/seoulawake/server/module/cafe/domain/Address.kt
package seoulawake.server.module.cafe.domain

import jakarta.persistence.Embeddable
import seoulawake.server.global.exception.ErrorCode
import seoulawake.server.global.exception.validate

@Embeddable
class Address(
  val address: String,
  val roadAddress: String,
) {
  init {
    validate(this.address.isNotBlank()) { ErrorCode.PARAMETER_VIOLATION_ERROR }
    validate(this.roadAddress.isNotBlank()) { ErrorCode.PARAMETER_VIOLATION_ERROR }
  }
}
